### PR TITLE
Add Use Cases page on auto-cancel

### PR DESCRIPTION
**Pull request by DamjanBecirovic**
Here is the separate page on the auto-cancel for Use Cases section.

For page title, I thought of `Auto-cancel pipelines on a new push`, but all other ideas are welcome.

### CODE REVIEW

File: auto-cancel.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
+The *auto-cancel* strategy defines the behavior of pipelines when they enter the
+queue for execution.
+
+By default, all pipelines will be queued and executed on a first-come, first-served
+basis.
+
+Often this is not the optimal solution, especially during development.
+
+We all head situations where you push some changes, only to realize that you have
```
Comment (shiroyasha) [inline auto-cancel.md]: typo, head -> had

File: auto-cancel.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+missed something small, so you push a new revision immediately and then find out
+another mistake...
+
+Before *auto-cancel* strategies the only way to get immediate feedback on the
+latest revision was to manually stop all pipelines from obsolete commits.
+
+Now if you set up an auto-cancel strategy this will be done automatically.
+That way you will get faster feedback on revisions that matter while skipping
+all the intermediate ones.
```
Comment (shiroyasha) [inline auto-cancel.md]: The temporal structure of `Before *auto-cancel* strategies` and `Now if you set up an auto-cancel` is more appropriate for blogs where you are announcing a new feature.

For documentation, it's better to keep the document time-neutral. Example would be:

Without auto-cancel...

If you set up auto-cancel...

File: auto-cancel.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+The *auto-cancel* strategy defines the behavior of pipelines when they enter the
+queue for execution.
+
+By default, all pipelines will be queued and executed on a first-come, first-served
+basis.
```
Comment (shiroyasha) [inline auto-cancel.md]: This is still true with the auto-cancel feature too. The FIFO order is not broken. Auto-cancel only "speeds up" the FIFO processing, and cancels redundant items.

File: auto-cancel.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+The execution of pipeline from commit A will be stopped and pipelines initiated
+from commits B and C that were waiting in the queue will be canceled.
+
+This will allow pipeline from newest commit D to start executing and provide you
+with feedback as soon as possible after a push.
+
+## Auto-cancel all pipelines on a new push to a non-master branch
+
+Often the auto-cancel strategy is ideal for feature branches during development.
```
Comment (shiroyasha) [inline auto-cancel.md]: I would drop the `Ofter the` part of the sentence. It is ok to be prescriptive in use cases.

File: auto-cancel.md (hunk 3 above)
Comment (DamjanBecirovic) [reply auto-cancel.md]: The later sentence should be related more with `all pipelines will be queued` part than FIFO, but I agree it is not so clear. I will rework that part.

**Comment by markoa** [comment]
Please post a comment after you do a round of edits, thanks.

**Comment by DamjanBecirovic** [comment]
@markoa edits are done.

File: auto-cancel.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
+The *auto-cancel* strategy defines the behavior of pipelines when they enter the
```
Comment (markoa) [inline auto-cancel.md]: Maybe rephrase this as "when a queue for execution forms"?

As is now it kind of sounds like you're describing any pipeline workflow in daily life...
